micrometer-core/src/test/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsConnectionPoolListenerTest.java

```diff
@@ -68,7 +73,8 @@ public void clusterOpening(ClusterOpeningEvent event) {
 
         assertThat(registry.get("mongodb.driver.pool.size").tags(tags).gauge().value()).isEqualTo(2);
         assertThat(registry.get("mongodb.driver.pool.checkedout").gauge().value()).isZero();
-        assertThat(registry.get("mongodb.driver.pool.waitqueuesize").gauge().value()).isZero();
+        // TODO: waitQueueEntered and waitQueueExited were removed, how can we provide mongodb.driver.pool.waitqueuesize?
```

Comment:
Do you happen to know anything from the Mongo team we could reference in our release notes? I asked about it in https://github.com/micrometer-metrics/micrometer/pull/2493#discussion_r590964124 before I saw your review comments.

Reply:
Not from the Mongo team per-se. However, [this deprecation list](https://mongodb.github.io/mongo-java-driver/3.12/javadoc/deprecated-list.html) (posted in 3.12) was super helpful when we migrated our apps to 4.x.

And [drilling down](https://mongodb.github.io/mongo-java-driver/3.12/javadoc/com/mongodb/event/ConnectionPoolWaitQueueEnteredEvent.html) directly into the event in question you see this loud warning:
| ⚠️  Deprecated. In the next major release, this event will no longer be published |
| --- |

Reply:
I see. We can certainly reference that. I was hoping for any reasoning why it was deprecated and removed, but lacking that, we can use the deprecation list you linked. As a user, did you ever use the wait queue metrics? Is there removal an issue for you or do you think it might be for other users?

Reply:
That JIRA issue is in connection with what I found: https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#why-are-waitqueuesize-and-waitqueuemultiple-deprecated

I'm going to ask the mongo community (`ConcurrentPool` has public `getCount`, `getInUseCount`, `getAvailableCount` but its interface does not and getting a reference could also be tricky.

Reply:
Except these are not available on master. :)
I asked this on the mongo forum, if you are interested: https://developer.mongodb.com/community/forums/t/java-monitoring-waitqueue-size/98845

Reply:
Between the deprecated list, release notes, and this spec outlining why the waitQueue metrics are deprecated it feels like the waitQueue metrics can be omitted.

Reply:
Between the deprecated list, release notes, and this spec outlining why the waitQueue metrics are deprecated it feels like the waitQueue metrics can be omitted.

Reply:
I meant to ask this earlier. How are docs handled in Micrometer? Do we need to open a "documentation ticket" for the removal of waitqueue metric?

Reply:
Depending on the outcome of the thread in the mongo forum, this can go to javadoc/release-notes/[micrometer-docs](https://github.com/micrometer-metrics/micrometer-docs).

Reply:
Good call in posting on that forum @jonatan-ivanov - looks like we got some direction. It does sound like this will give roughly the same as waitqueue. I understand metrics are APIs and its nice to not break consumers. However, I feel like we should deprecate that metric as Mongo does not emit it anymore. What are your thoughts Jonatan and @shakuzen ? 

Also, what is that process like for Micrometer metrics?

Reply:
I don't think any of the metrics backends we support have a concept of a deprecated metric (though it is an interesting idea), so unfortunately we're limited by what we can do to inform users. Release notes are typically where we would call out a removal. I agree we should consider whether it is an important metric for users or not. It seems the Mongo team thinks it isn't something to be tracked, judging from their removal of it. If there are Micrometer and MongoDB users out there that find it useful, we'd like to hear about that. We could drop an ask in the release notes and remove it in the next minor version if we don't hear people are using it.